File: ApplicationService/Services/DashboardService.cs
using ApplicationService.IServices;
using ApplicationService.Utilities;
using DataRepository.EntityModels;
using DataRepository.IRepository;
using DataRepository.Repositoryy;
using Microsoft.Extensions.Configuration;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ApplicationService.Services
{
    public class DashboardService:IDashboardService
    {
        private readonly IDashboardRepository _dashboardRepository;
        private readonly IConfiguration _config;
        public DashboardService(IDashboardRepository dashboardRepository,IConfiguration config)
        {
            _dashboardRepository = dashboardRepository;
            _config = config;
        }

        public Task<LinechartData> GetAllTicketCreated(string startDate, string endDate, string userId, int companyId)
        {
           return _dashboardRepository.GetAllTicketCreated(startDate, endDate,userId,companyId);
        }

        public Task<List<PrioritChartResponse>> GetAllTicketsWithPriority(string userId, string userType, int companyId)
        {
            return _dashboardRepository.GetAllTicketsWithPriority(userId,userType, companyId);
        }

        public ResponseStatus GetCompanyCount()
        {
           return _dashboardRepository.GetCompanyCount();
        }
        public Task<DashboardResponseStatus> GetUserAndTicketCount(string userId, int companyId)
        {
            return _dashboardRepository.GetUserAndTicketCount(userId, companyId);
        }
        public ChartResponse GetChartDataByDepartment(string userId, string userType, int companyId)
        {
            return _dashboardRepository.GetChartDataByDepartment(userId,userType ,companyId);
        }

    }
}

File: HelpDesk_TicketSystem/ClientApp/src/services/dashboardService/dashboard.service.ts
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { environment } from '../../environments/environment';

@Injectable({
  providedIn: 'root'
})
export class DashboardService {

  constructor(private http: HttpClient) {
  }
  apiUrl = environment.apiBaseUrl;

  GetCompanyCount() {
    let url = `${this.apiUrl}Dashboard/GetCompanyCount`;
    return this.http.get(url);
  }

    GetUserAndTicketCount(userId: string, companyId: number) {
      let url = `${this.apiUrl}Dashboard/GetUserAndTicketCount/${userId}/${companyId}`;
      return this.http.get(url);
  }

  GetChartDataByDepartment(userId: string, userType: string ,companyId: number) {
    let url = `${this.apiUrl}Dashboard/GetChartDataByDepartment/${userId}/${userType}/${companyId}`;
      return this.http.get(url);
    }
  GetAllTicketsWithPriority(userId: string, userType: string,companyId: number) {
    let url = `${this.apiUrl}Dashboard/GetAllTicketsWithPriority/${userId}/${userType}/${companyId}`;
    return this.http.get(url);
  }
  GetAllTicketCreated(startDate: string, endDate: string, userId: string, companyId: number) {
    let url = `${this.apiUrl}Dashboard/GetAllTicketCreated/${userId}/${companyId}/${startDate}/${endDate}`;
    return this.http.get(url);
  }
}
